hw/ip/otbn/dv/otbnsim/sim/dmem.py

```diff
@@ -48,82 +48,25 @@ def __init__(self) -> None:
             raise RuntimeError('DMEM size ({}) is not divisible by 32.'
                                .format(dmem_size))
 
-        num_words = dmem_size // 32
+        num_words = dmem_size // 4
 
         # Initialise the memory to an arbitrary "bad" constant (here,
         # 0xdeadbeef). We could initialise to a random value, but maybe it's
         # more helpful to generate something recognisable for now.
-        uninit = 0
-        for i in range(32 // 4):
-            uninit = (uninit << 32) | 0xdeadbeef
+        uninit = 0xdeadbeef
 
         self.data = [uninit] * num_words
         self.trace = []  # type: List[TraceDmemStore]
 
-    def _get_u32s(self, idx: int) -> List[int]:
-        '''Return the value at idx as 8 uint32's
-
-        These are ordered by increasing address, so the first word from
-        _get_u32s(0) will correspond to bytes at addresses 0x0 through 0x3.
-
-        These uint32's are the words that get loaded by word accesses to
-        memory. Since these accesses are also little-endian, the byte at memory
-        address 0x0 (which holds the LSB for the first 256-bit value) will also
-        be the LSB of the first u32 returned by _get_u32s(0).
-
-        '''
-        assert 0 <= idx < len(self.data)
-
-        word = self.data[idx]
-        assert 0 <= word <= 1 << 256
-
-        # Pack this unsigned word into w32s as 8 32-bit numbers. Note that
-        # this packing is "little-endian": the first unsigned word contains
-        # the LSB of the value.
-        ret = []
-        w32_mask = (1 << 32) - 1
-        for subidx in range(8):
-            ret.append((word >> (32 * subidx)) & w32_mask)
-
-        return ret
-
-    def _set_u32s(self, idx: int, u32s: List[int]) -> None:
-        '''Set the value at idx with 8 uint32's in little-endian order'''
-        assert 0 <= idx < len(self.data)
-        assert len(u32s) == 8
-
-        # Accumulate the u32s into a 256-bit unsigned number (in a
-        # little-endian fashion)
-        u256 = 0
-        for u32 in reversed(u32s):
-            assert 0 <= u32 <= (1 << 32) - 1
-            u256 = (u256 << 32) | u32
-
-        # Store it!
-        self.data[idx] = u256
-
     def load_le_words(self, data: bytes) -> None:
         '''Replace the start of memory with data'''
         if len(data) > 32 * len(self.data):
             raise ValueError('Trying to load {} bytes of data, but DMEM '
                              'is only {} bytes long.'
                              .format(len(data), 32 * len(self.data)))
 
-        # Zero-pad bytes up to the next multiple of 256 bits (because things
```

Comment:
Is it guaranteed that the incoming data is a multiple of 32bits? If not then something similar to this will still be needed to pad up to the next 32bit boundary (it looks like an arbitrary byte array)

Reply:
It's not guaranteed, I don't think.

Reply:
Made necessary changes